### PR TITLE
Adds a setter for density

### PR DESCRIPTION
**Pull request by Rohesie**
## About The Pull Request

* Adds set_density()
* Fixes one instance of a duplicate density assignment on an object.
* Comments two hacky usages of density which will have to forgo using the setter for now.

## Why It's Good For The Game

Lets us append code to the event of density changing.
Should be properly self-contained and easy to review.

## Changelog
No player-facing changes.

### CODE REVIEW

File: code/modules/mob/mob.dm [hunk 1]
```diff
@@ -969,9 +969,9 @@
 	var/turf/T = get_turf(src)
 	if(M.loc != T)
 		var/old_density = density
-		density = FALSE
+		density = FALSE // Hacky and doesn't use set_density()
```
Comment (MrDoomBringer) [inline code/modules/mob/mob.dm:972]: Thoughts on replacing manual changes like this with a quick_set_density(FALSE) macro? I'm not terribly thrilled about copy+pasting this comment wherever needed

**Review by MrDoomBringer — CHANGES_REQUESTED** [review]

File: code/modules/mob/mob.dm (hunk 1 above)
Comment (MrDoomBringer) [reply code/modules/mob/mob.dm:972]: (Unless we plan on removing the need for this entirely in the near future)

Comment (Rohesie) [reply code/modules/mob/mob.dm:972]: The long-term idea is to remove this if possible, and not expand on it.
It happens in two places in the code only, both commented.

**Comment by optimumtact** [comment]
setters are explicitly banned in our contributing guidelines

**Comment by Rohesie** [comment]
Setters good, it's getters that are (sometimes) bad.

**Comment by Rohesie** [comment]
🗞️

**Pull request merged**